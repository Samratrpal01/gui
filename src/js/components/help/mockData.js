export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {},
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '01-01-2003'
  }
};
