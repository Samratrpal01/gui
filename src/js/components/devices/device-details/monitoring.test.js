import React from 'react';
import { prettyDOM } from '@testing-library/dom';
import { render } from '@testing-library/react';
import DeviceMonitoring, { DeviceMonitorsMissingNote } from './monitoring';
import { defaultState, undefineds } from '../../../../../tests/mockData';

describe('tiny components', () => {
  [DeviceMonitorsMissingNote].forEach(async Component => {
    it(`renders ${Component.displayName || Component.name} correctly`, () => {
      const { baseElement } = render(<Component docsVersion="" />);
      const view = baseElement.firstChild;
      expect(view).toMatchSnapshot();
      expect(view).toEqual(expect.not.stringMatching(undefineds));
    });
  });
});

describe('DeviceMonitoring Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(
      <DeviceMonitoring
        alerts={defaultState.monitor.alerts.byDeviceId.a1}
        alertListState={{ page: 2, perPage: 20, total: 9001 }}
        device={defaultState.devices.byId.a1}
        getAlerts={jest.fn}
        isOffline
        latestAlerts={defaultState.monitor.alerts.byDeviceId.a1}
        setAlertListState={jest.fn}
      />
    );
    // special snapshot handling here to work around unstable ids in mui code...
    const view = prettyDOM(baseElement.firstChild.firstChild, 100000, { highlight: false })
      .replace(/id="mui-[0-9]*"/g, '')
      .replace(/aria-labelledby="(mui-[0-9]* *)*"/g, '')
      .replace(/\\/g, '');
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
