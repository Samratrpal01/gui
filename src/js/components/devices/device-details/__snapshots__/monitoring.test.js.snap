// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceMonitoring Component renders correctly 1`] = `
<div
  class="margin-bottom "
>
  <div
    class="clickable flexbox space-between"
    style="align-items: flex-start;"
  >
    <div
      class="flexbox center-aligned"
    >
      <h4
        class="margin-right"
      >
        Monitoring
      </h4>
    </div>
    <div
      class="flexbox centered"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
        disabled=""
        style="margin-right: 16px; font-weight: 900; font-size: smaller;"
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Add-on
        </span>
      </button>
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div>
    <div
      class="flexbox center-aligned padding-small device-detail-notification  red column-data "
      style="margin-bottom: 8px; padding-bottom: 12px; padding-top: 12px;"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root red"
        focusable="false"
        style="margin-right: 16px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
        />
      </svg>
      <div
        class="flexbox center-aligned"
      >
        <div
          class="key text-muted margin-right-small"
        >
          <b>
            Device offline
          </b>
        </div>
        Last check-in over 
        24
         
        hours
         ago
      </div>
    </div>
  </div>
  <div
    class="MuiCollapse-container fadeIn"
    style="min-height: 0px; height: 0px; transition-duration: 0ms;"
  >
    <div
      class="MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner"
      >
        <div
          class="margin-bottom"
        >
          <h4
            class="text-muted"
          >
            Triggered alerts
          </h4>
          <div
            class="monitoring-alert column-data"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root red"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
              />
            </svg>
            <div
              class="key text-muted"
            >
              <b>
                SSH Daemon is not running
              </b>
            </div>
            <div>
              CRITICAL
            </div>
            <time
              datetime="2021-07-23T12:22:36+00:00"
            >
              2021-07-23 12:22
            </time>
            <a>
              view 
              details
            </a>
          </div>
        </div>
        <div
          class="margin-top-small"
        >
          <a>
            show less
          </a>
        </div>
      </div>
    </div>
  </div>
  <hr
    class="MuiDivider-root"
    style="margin-top: 24px;"
  />
</div>
`;
